app/views/users/_profile.html.erb

```diff
@@ -60,7 +60,10 @@
       <% group.profile_fields.each do |field| %>
         <div class="crayons-field <%= "crayons-field--checkbox" if field.input_type == "check_box" %>">
           <% if field["input_type"] == "check_box" %>
-            <%= f.public_send(field["input_type"], "profile[#{field.attribute_name}]", checked: @user.profile.data[field.attribute_name], class: "crayons-checkbox") %>
+            <%= f.public_send(field["input_type"],
+                              "profile[#{field.attribute_name}]",
+                              checked: profile_field_value(@user.profile, field),
```

Comment:
See the longer explanation above, we shouldn't need a helper we just need to go through the profile instance, not the data hash directly:

```ruby
@user.profile.public_send(field.attribute_name)
```